### PR TITLE
Packet handling

### PR DESCRIPTION
**Pull request by daniel-lerch**
This pull request adds the code structure for VSL and packet handling.
There are still many features missing, concurrency conflicts, etc. but other parts of the project need this code for continuation of development.

### CODE REVIEW

File: src/SkynetServer/Network/Client.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        private Session session;
+
+        public async Task SendPacket(Packet packet)
+        {
+            using (var buffer = PacketBuffer.CreateDynamic())
+            {
+                packet.WritePacket(buffer);
+                bool success = await socket.SendPacketAsync(packet.Id, buffer.ToArray());
+                if (!success)
+                    Console.WriteLine($"Failed to send packet {packet.Id}");
```
Comment (Twometer) [inline src/SkynetServer/Network/Client.cs]: shouldn we close the connection here?

File: src/SkynetServer/Network/Packet.cs [hunk 2]
```diff
@@ -11,30 +11,42 @@ internal abstract class Packet
...
+                    instance.Policy = attribute.PacketPolicy;
+                    packets.Add(instance);
+                    if (attribute.PacketId > max) max = attribute.PacketId;
+                }
+            }
 
             Packets = new Packet[max + 1];
 
-            foreach ((Type type, PacketAttribute attribute) in packets)
+            foreach (Packet packet in packets)
```
Comment (Twometer) [inline src/SkynetServer/Network/Packet.cs]: Why not just .ToArray()?

File: src/SkynetServer/appsettings.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
...fqJA/uFtpiYL0UYJwQgLhwYwes8hDTqb9o4k+QjOHZl59381lvhHD92LoywZvRyUBVFXzTKAvx9/AhziaZxXX2dyv3Zic+M=</P><Q>wQRaLGTiutc/Z+yW4LfW8LO2kDxclJ8+bfyI9bxxF4myr3qtWSKJUh4pkaiPxSgiq6XDKmei/kAN0cZevIUplDEVtdQ+1WwaZtzCKeJEr9regyhPRDt8I0uCPIV8l7YqA84GPtEnFHiz88WiHaBcDNK2SUaj96Uf8pDrcO0OfkU=</Q><DP>bc/BhOsIIZ4Z529xn/w/5n1OAxmtgAKSxLn38mQK8Jq8dWyq1jAYU0AQaWUtHGxahQdBJe/JFWWWlysL1Lp9ntFAHXL7JMqGdnh3KqImUzIPQQWTz52WLKf8uWf/9Be5RCZkHaxEjiJ3IdYqNxal31k7mdPElOM4Lmimivm0o38=</DP><DQ>sGH6yVDUBKvIrH5+jhJRoBqKPZp7e/FdyoKY4f73VewmVseDe83csScRE6N25lM8A6FyDOEyCKeeOVHOpvYFPf/w9/1iAgRMyCz6Xe4AmLco2bBPZ2n4kgRhmk7dwYHdOSbffC4DSWjYAW9cKzishadUnvenxRmnbTHThZmLwgU=</DQ><InverseQ>IE7ehnkv7kYY+w5s6bCgy0cFSi4EZNG52emEntjEt0DuULADptmTzoPb5E4+JPmD8jZWK6Mr3QbOdtRBXvqak+Hr8u2yjg+AgImfgpLSeG9UtrvAKpnMv132vy6vQYmT5wKX22XYo3TplWJoMv9yMB8v7LM1aGNKcCjpRBKofPI=</InverseQ><D>AAwgwJ7HpCw6sEmFoLXK29diOJgGa3is+MiAvJSCEVmJz4qYlMo3+RRA7qAANLSxASkDtIoVHtizz6rANfp/CbKCtJ7ANa6FxW9GM01fBUNii9nd48jkhZ5nR0TotGGUhCWmDxbxK9ExiLL13SEsaRpDaer6sdG8QIZwG4SU6N69YO40/QXYN2OhzXGs5j1ZjA6sLSzcfRC7/yRnmdG2CSbnuVghue+YdAqf9vH2gyD1aDJhqajzGoVc5XCmu7JTX5fI1EruVUX0LATW5966CRPj6RMgEo8MadyS0M+QdF+Et2jzmCj3NvnbkxegvpBvbSFEVgVT8LbbG2ro+7yzfQ==</D></RSAKeyValue>"
```
Comment (Twometer) [inline src/SkynetServer/appsettings.json]: I hope that this is just a demo key ;-)

File: src/SkynetServer/Network/ProtocolException.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SkynetServer.Network
+{
+    internal class ProtocolException : Exception
+    {
+        public ProtocolException() : this("Client violation of Skynet Protocol rules caused an exception") { }
```
Comment (Twometer) [inline src/SkynetServer/Network/ProtocolException.cs]: I think that 'caused an exception' can be left out, as that should be pretty obvious at this point

File: src/SkynetServer/appsettings.json (hunk 3 above)
Comment (daniel-lerch) [reply src/SkynetServer/appsettings.json]: This our debug keypair which should be used in all clients when connecting to debug servers.

File: src/SkynetServer/Network/Client.cs (hunk 1 above)
Comment (daniel-lerch) [reply src/SkynetServer/Network/Client.cs]: VSL will automatically call `IVSLCallback.OnConnectionClosed()` when it fails to send a message. The Console output is only for debugging purposes like everywhere else as we don't have a logging API yet.

**Pull request merged**